tests/dma/vif/unpack.cpp

```diff
@@ -0,0 +1,131 @@
+#include <common-ee.h>
+#include <kernel.h>
+#include <string.h>
+#include "../dmaregs.h"
+#include "../dmasend.h"
+#include "../dmatags.h"
+#include "emit_vifcode.h"
+#include "vifunits.h"
+
+void testSimpleUnpack(VIF::Unit *unit, const char *description, VIF::UnpackType unpackType, bool zeroExtend, u32 value0, u32 value1 = 0, u32 value2 = 0, u32 value3 = 0) {
+	unit->regs->fbrst = VIF::FBRST_RST;
+	
+	static const u32 packetSize = 16 * 2;
+	static const u32 vectorSize = 16 * 1;
+	
+	u8 *vuMem = unit->vuMem;
+	volatile u32 *vu_32 = (volatile u32 *)vuMem;
+	
+	memset(vuMem, 0xFF, vectorSize);
+	SyncDCache(vuMem, vuMem + vectorSize);
+	
+	VIF::Packet vifcode(packetSize);
+	vifcode.STCYCL(4, 4);
+	vifcode.UNPACK(unpackType, 1, 0, zeroExtend ? VIF::UNPACK_ZERO_EXTEND : VIF::UNPACK_NORMAL);
+	vifcode.Data32(value0);
+	vifcode.Data32(value1);
+	vifcode.Data32(value2);
+	vifcode.Data32(value3);
+	
+	DMA::SendSimple(unit->dmaChannel, vifcode.Raw(), packetSize);
+	
+	SyncDCache(vuMem, vuMem + vectorSize);
+	
+	printf("Simple UNPACK %s - %08x - %08x - %08x - %08x\n", 
+		description, vu_32[0], vu_32[1], vu_32[2], vu_32[3]);
+}
+
+void testSetUnpack(VIF::Unit *unit) {
+	unit->regs->fbrst = VIF::FBRST_RST;
+	
+	static const u32 packetSize = 16 * 4;
+	static const u32 vectorSize = 16 * 256;
+	
+	u8 *vuMem = unit->vuMem;
+	volatile u32 *vu_32 = (volatile u32 *)vuMem;
+	
+	memset(vuMem, 0xFF, vectorSize);
+	SyncDCache(vuMem, vuMem + vectorSize);
+	
+	VIF::Packet vifcode(packetSize);
+	vifcode.STCOL(0xFFEEDDCC, 0xBBAA9988, 0x77665544, 0x33221100);
+	vifcode.STMASK(0xAAAAAAAA); //Use COL value for all writes
+	vifcode.STCYCL(1, 0);
+	vifcode.UNPACK(VIF::UNPACK_TYPE_S8, 256, 0, VIF::UNPACK_ENABLE_MASKS);
+	
+	DMA::SendSimple(unit->dmaChannel, vifcode.Raw(), packetSize);
+	
+	SyncDCache(vuMem, vuMem + vectorSize);
+	
+	printf("Set 4096 bytes UNPACK - %08x - %08x - %08x - %08x\n", 
+		vu_32[0], vu_32[1], vu_32[1022], vu_32[1023]);
+}
+
+void testNoDataUnpack(VIF::Unit *unit, u32 num) {
+	unit->regs->fbrst = VIF::FBRST_RST;
+	
+	u32 vpsBefore = unit->regs->stat & VIF::STAT_VPS_MASK;
+	
+	static const u32 packetSize = 16;
+	
+	VIF::Packet vifcode(packetSize);
+	vifcode.NOP();
+	vifcode.NOP();
+	vifcode.STCYCL(4, 4);
+	vifcode.UNPACK(VIF::UNPACK_TYPE_S8, num, 0);
+	
+	DMA::SendSimple(unit->dmaChannel, vifcode.Raw(), packetSize);
+	
+	u32 vpsAfter = unit->regs->stat & VIF::STAT_VPS_MASK;
+	
+	printf("No data UNPACK(NUM: %d) -> VPS before: %d, VPS after: %d, CODE: %08x, NUM: %d\n", 
+	    num, vpsBefore, vpsAfter, unit->regs->code, unit->regs->num);
```

Comment:
Does this actually set the memory to anything?  Or just adjust the other regs?

-[Unknown]